arch: arm: core: fix vector table relocate write to flash

b0661 opened the pull request:
Hello Maciej,

this pull request contains the fix for the vector table relocation.
There are some additional commits from zephyr master that should vanish when you update your branch.

Hope that helps
Bobby

MaciejDebski commented:
@b0661 
Could you possibly rebase your code on the newest version of my code? I should have merge your pull request first, I am sorry. I updated the hal/ll library to 1.9.
I will merge it tomorrow or as you manage to resolve conflicts. Take your time.

Thank you,
Maciej Dębski

b0661 committed:
arch: arm: core: fix vector table relocate write to flash

Some SOCs (e.g. STM32F0) can map the flash to address 0 and
the flash base address at the same time. Prevent writing to
duplicate flash address which stops the SOC.

Allow Cortex M SOCs to create their own vector table relocation
function.

Provide a relocation function for STM32F0x SOCs.

Signed-off-by: Bobby Noelte <b0661n0e17e@gmail.com>

b0661 commented:
@MaciejDebski 
Now rebased. Only did a compile test. Hope that works with 1.9.

Bobby
